backtrace-library/src/main/cpp/backtrace-native.cpp

```diff
@@ -195,15 +195,15 @@ void Crash() {
     *(volatile int *) 0 = 0;
 }
 
-void DumpWithoutCrash(jstring message) {
+void DumpWithoutCrash(jstring message, jboolean set_main_thread_as_faulting_thread) {
     crashpad::NativeCPUContext context;
     crashpad::CaptureContext(&context);
 
     // set dump message for single report
     crashpad::SimpleStringDictionary *annotations = NULL;
     bool did_attach = false;
 
-    if (message != NULL) {
+    if (message != NULL || set_main_thread_as_faulting_thread == true) {
```

Comment:
These checks and behaviors seem like they are independent. Maybe we can break up this if statement?